torch/_dynamo/testing.py

```diff
@@ -46,7 +46,7 @@ def remove_optimized_module_prefix(name):
     prefix = "_orig_mod."
     assert name.startswith(prefix)
     name = name[len(prefix) :]
-    return torch.distributed.fsdp._common_utils.clean_tensor_name(name)
+    return name
```

Comment:
@wconstab

Context: This PR changes FSDP so that when `use_orig_params=True`, the parameter names returned from `named_parameters()` include the `_fsdp_wrapped_module.` prefixes. Before, FSDP overrode `named_parameters()` to clean the prefix (see `fully_sharded_data_parallel.py`). I found that overriding it to clean the prefix creates unwanted discrepancy with the true module structure, which can lead to some issues. We can leave having clean parameter names to the currently-developed composable APIs (e.g. `fully_shard`).

This change here in `_dynamo/testing.py` to not call `clean_tensor_name()` allows the unit tests to pass. I still wanted to check: does Dynamo + FSDP rely on FSDP's `named_parameters()` returning the cleaned names (i.e. without `_fsdp_wrapped_module.`)?

Reply:
I don't think dynamo relies on the structure of the name either way.  other than we do rely on finding the "is_fsdp_wrapped_module" boolean for enablement of fsdp handling.


question: what is the history of `remove_optimized_module_prefix` fn?  at a glance, i have no memory of modifying this myself for fsdp but i see it had an fsdp clean param thing in it.  did you put that there before and are taking it out now?

Reply:
`removed_optimized_module_prefix` came from https://github.com/pytorch/pytorch/pull/89113

My understanding is as follows:
https://github.com/pytorch/pytorch/blob/46a81c8db786ac03abc16bbecc8401dcbf451f10/torch/_dynamo/testing.py#L63-L65
https://github.com/pytorch/pytorch/blob/46a81c8db786ac03abc16bbecc8401dcbf451f10/test/distributed/test_dynamo_distributed.py#L368-L370

The `eager_model` removed the prefixes but the `opt_model` did not, so for parity, we needed to manually remove the prefixes for `opt_model`. However, with this PR, `eager_model` no longer removes the prefixes, so similarly, we can remove the manual removal for `opt_model`.

In this case, we should be safe to land this.

Reply:
seems ok to me